core/lib/src/config/secret_key.rs

```diff
@@ -1,15 +1,15 @@
 use std::fmt;
 
-use aes_gcm::{
-    AeadCore, Aes256Gcm, Nonce,
-    aead::{generic_array::GenericArray, Aead, KeyInit, OsRng},
+use chacha20poly1305::{
+    aead::{Aead, AeadCore, KeyInit, OsRng, generic_array::GenericArray},
+    XChaCha20Poly1305, XNonce,
 };
 use cookie::Key;
 use serde::{de, ser, Deserialize, Serialize};
 
 use crate::request::{Outcome, Request, FromRequest};
 
-const NONCE_LEN: usize = 12;
+const NONCE_LEN: usize = 24; // 192-bit
 const KEY_LEN: usize = 32;
```

Comment:
Same with this. We should be able to get it from the [`Key`](https://docs.rs/chacha20poly1305/latest/chacha20poly1305/type.Key.html) and [`ConstArrayLen`](https://docs.rs/generic-array/latest/generic_array/type.ConstArrayLength.html).